### PR TITLE
Rewrite number literal parsers

### LINKED ISSUE #108: Capital letters break hexadecimal, binary numeric literals
`a = 0b001101` and `a = 0x3fff` are parsed appropriately with the binary and hexadecimal showing up as a numeric literal.

However, `a = 0B001101` and `a = 0X3FFF` will show the beginning 0 as a numeric, and the rest will break.

### PR DESCRIPTION
**Pull request by elbeno**
Addresses #224

### CODE REVIEW

**Comment by elbeno** [comment]
Suggestions for tests? There are a lot of combinations...

**Comment by amaanq** [comment]
just one function with whatever combo of expression statements you can think of is good

```c
int main() {
    0x204;
    10;
    // etc..
}
```

**Comment by elbeno** [comment]
OK, I added a bunch.

File: grammar.js [hunk 1]
```diff
@@ -1247,6 +1247,60 @@ module.exports = grammar(C, {
...
+              hexDigits,
+              seq(hexDigits, '.', optional(hexDigits)),
+              seq('.', hexDigits)),
+            hexExponent),
+        ),
+        optional(suffix),
+      ));
+    },
+
+    number_literal: $ => choice($._integer_literal, $._float_literal),
```
Comment (amaanq) [inline grammar.js:1302]: Can we just keep like how C does it? Just a single block would be preferred to encapsulate both cases

Comment (elbeno) [reply grammar.js:1302]: Integer and float literals have pretty different rules, which is why I split them up. Trying to do them in one block is probably why they're currently inaccurate...

Comment (amaanq) [reply grammar.js:1302]: if they're hidden it wouldn't matter though, in the main number block you can just return a choice of either

Comment (elbeno) [reply grammar.js:1302]: Oh I see what you're suggesting.

**Comment by amaanq** [comment]
Also need to rebase

**Comment by amaanq** [comment]
I fixed #108 in C and updated the grammar here

**Comment by amaanq** [comment]
Thanks, that looks a lot nicer

**Pull request merged**